src/Plugin/Filter/HtmlPurifierFilter.php

```diff
@@ -55,8 +24,15 @@ public function process($text, $langcode) {
     /* @var $config \HTMLPurifier_Config */
     $config = \HTMLPurifier_Config::createDefault();
 
-    // Get and apply the configuration stored in Drupal.
-    $this->applyConfigSettings($config, $this->drupalConfig);
+    // Get and apply the configuration set in the filter form.
+    if (!empty($this->settings['htmlpurifier_configuration'])) {
+      $settings = Yaml::decode($this->settings['htmlpurifier_configuration']);
+      foreach ($settings as $namespace => $directives) {
+        foreach ($directives as $key => $value) {
+          $config->set("$namespace.$key", $value);
```

Comment:
The only downside that I see with this approach is that as soon that you save this configuration Drupal will try to validate it and will fail. You can double check that by installing config_inspector module and search for `htmlpurifier_configuration`.

Perhaps an alternative to overcome this is to store the configuration as a JSON string into  htmlpurifier.settings.plugin_config and use json_encode and json_decode before saving and showing it in a textarea respectivelly.

Reply:
If I'm understanding what you're saying I hope this config schema would be the missing piece: https://github.com/juanolalla/htmlpurifier/pull/11/commits/bc36e9ede57b017b02635d771b1a988352ecd2f3. Although I don't know how to test that failure you're talking about would be gone.

Not sure what would be the purpose for json_encode the already Yaml::encode(), which is a string anyway, and added like that to the schema?

Reply:
The diff looks like `$config` is a Drupal configuration setting, even though it's actually an HTML Purifier object. That's probably why @juampynr mentioned config schemas.

Perhaps we rename this to `$purifierConfig`?

Reply:
Then, do we need a config schema for the settingsForm element or not?